tests/js/server/aql/aql-optimizer-rule-decay-unnecessary-sorted-gather.js

```diff
@@ -0,0 +1,199 @@
+/*jshint globalstrict:false, strict:false, maxlen: 500 */
+/*global assertTrue, assertEqual, assertNotEqual, AQL_EXPLAIN, AQL_EXECUTE */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief tests for optimizer rules
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2014 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+/// @author Copyright 2020, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const db = require("@arangodb").db;
+const jsunity = require("jsunity");
+const console = require('console');
+const _ = require("lodash");
+const isEnterprise = require("internal").isEnterprise();
+
+function optimizerRuleTestSuite() {
+  const ruleName = "decay-unnecessary-sorted-gather";
+
+  const sharedCollection = "c";
+  const singleShardCollection = "d";
+
+  const satelliteCollection = "sat";
+
+  const smartJoinCollection1 = "smartJoin1";
+  const smartJoinCollection2 = "smartJoin2";
+  const smartJoinAttribute = "attr";
+
+  return {
+
+    setUpAll: function () {
+      db._create(sharedCollection, {numberOfShards: 3});
+      db._create(singleShardCollection, {numberOfShards: 1});
+
+      if (isEnterprise) {
+        db._create(satelliteCollection, {replicationFactor: "satellite"});
+
+        db._create(smartJoinCollection1, {numberOfShards: 6, shardKeys: ["_key"]});
+        db._create(smartJoinCollection2, {numberOfShards: 6, shardKeys: ["_key:"], smartJoinAttribute: smartJoinAttribute, distributeShardsLike: smartJoinCollection1});
+
+      }
+    },
+
+    tearDownAll: function () {
+      db._drop(sharedCollection);
+      db._drop(singleShardCollection);
+      db._drop(satelliteCollection);
+      db._drop(smartJoinCollection2);
+      db._drop(smartJoinCollection1);
+    },
+
+    testNonEnterprise: function () {
+      // this query has to be that complicated, because
+      // 1. if we only use a single collection with repl fact 1, the one shard rule will apply
+      // 2. the FILTER forces the sharded collection to enumerated first
```

Comment:
You could disable the `interchange-adjacent-enumeration` rule, which I suppose switched the enumerations.